components/gitpod-protocol/src/util/tracing.ts

```diff
@@ -138,13 +138,21 @@ export namespace TraceContext {
         }
         const namespace = _namespace ? `${_namespace}.` : '';
 
-        for (const k of Object.keys(keyValueMap)) {
-            const v = keyValueMap[k];
-            if (typeof v === 'object') {
-                addNestedTags(ctx, v, `${namespace}${k}`);
-            } else {
-                ctx.span.setTag(`${namespace}${k}`, v);
+        try {
+            for (const k of Object.keys(keyValueMap)) {
+                const v = keyValueMap[k];
+                // oh my JavaScript... `typeof null` and `typeof undefined` return `object`
+                if (v !== null
+                    && v !== undefined
+                    && typeof v === 'object') {
```

Comment:
I think you could also use `instanceof`, which is generally considered much safer than `typeof`:

```suggestion
                if (v instanceof Object) {
```

<img width="244" alt="Screenshot 2021-12-08 at 12 08 19" src="https://user-images.githubusercontent.com/599268/145198456-6750bc13-7333-437a-b2d9-ab318313cac9.png">

Reply:
Thx @jankeromnes ! My :brain: refuses to learn this JS `intricacies` so I'm doomed to re-learn every other year. :laughing: